compiler/rustc_data_structures/src/sync.rs

```diff
@@ -295,8 +295,8 @@ cfg_if! {
         pub auto trait DynSend {}
         pub auto trait DynSync {}
     } else {
-        pub use std::marker::Send as Send;
-        pub use std::marker::Sync as Sync;
+        pub use DynSend as Send;
+        pub use DynSync as Sync;
```

Comment:
These reexports should be removed. `DynSend` is not the same as `Send`.

Reply:
That makes sense!